### DISCUSSION TITLE
How to accept diesel table and column as argument?

### DISCUSSION THREAD

**Discussion by ghost** [diesel-rs/diesel#3355] [Q&A]
In my rust code, I want to write generic function that gets wilko by public_id. Here is what I have

I have 3 tables. They all have the same field and structure.

```sql
CREATE TABLE "wilko"
(
    "id"         serial primary key,
    "public_id" integer NOT NULL
);
```

```rs
use diesel::query_dsl::filter_dsl::FilterDsl;
use diesel::query_dsl::methods::LimitDsl;
use diesel::sql_types::{Bool, Integer, SingleValue, SqlType, Text};
use diesel::{dsl, Column, Expression, Table, TextExpressionMethods, PgConnection};
use serde::{Deserialize, Serialize};


#[derive(
    diesel::Queryable,
    diesel::Identifiable,
    diesel::Selectable,
    Debug,
    PartialEq,
    Deserialize,
    Serialize,
    Clone,
)]
#[diesel(table_name = crate::schema::wilko)]
pub struct Wilko {
    pub id: i32,
    pub public_id: i32,
}

pub async fn get_wall<T, C, R: Send>(
    table: T,
    column: C,
    public_key: String,
    mut conn: PgConnection,
) -> R
where
    T: Table + FilterDsl<dsl::Eq<C, Integer>>,
    C: Column + Expression<SqlType = Integer>,
    dsl::Filter<T, dsl::Eq<C, Integer>>: LimitDsl,
    <T as FilterDsl<dsl::Eq<C, Integer>>>::Output: Serialize,
{
    Ok(diesel_async::RunQueryDsl::get_result::<Wilko>(
        table.filter(column.eq(public_key)),
        &mut conn,
    )
    .await.unwrap())
}
```

Helpful links

[rust - Pass any Diesel table as an argument - Stack Overflow](https://stackoverflow.com/questions/64120177/pass-any-diesel-table-as-an-argument)

Using diesel 2.0

**Comment by weiznich** [comment]
You code mixes `diesel` and `diesel_async` code in an incompatible way. As already stated: That's the diesel discussion forum here, so we only provide support for `diesel` itself here. 
As for `diesel_async` questions: Please use the discussion forum in the corresponding repository to ask questions there. Please provide a link to the full source code as context if you ask questions there.
